### DISCUSSION TITLE
How do I create public profile for users on signup?

### DISCUSSION THREAD

**Discussion by xt42io** [supabase/supabase#9717] [Questions]
I set up a "login with GitHub" authentication method with Supabase which works, but I need to have the users on a public profile table. I followed some instructions:
[https://daily-dev-tips.com/posts/supabase-automatically-create-user-profiles-on-sign-up/](https://daily-dev-tips.com/posts/supabase-automatically-create-user-profiles-on-sign-up/)
[https://github.com/supabase/supabase/issues/563](https://github.com/supabase/supabase/issues/563)

When I followed the steps @kiwicopple provided, it worked but that was just for the id and email...when I tried to add my own additional information (id, email, display name and avatar URL) for the data I needed, it didn't work and I got this error ` database error saving new user`

I also saw in the discussions that the table should only be created with SQL and not the GUI which I did but it still didn't work...

Here is my code 👇🏽

`create function public.handle_new_user()
returns trigger
language plpgsql
security definer set search_path = public
as $$
begin
  insert into public.users (id, email, username, avatar_url, display_name)
  values (new.id, new.raw_user_meta_data ->> 'user_name',  new.raw_user_meta_data ->> 'email',  new.raw_user_meta_data ->> 'avatar_url', new.raw_user_meta_data.name);
  return new;
end;
$$;`

Help!

**Comment by GaryAustin1** [comment]
You are probably not pulling the data correctly from the auth.users row provided to the trigger function, the data does not exist in the inserted row, or you have a mismatch in data types to the table you are inserting too.

You might be able to see the reason in the dashboard database tab, then use the Postgres log right after you attempt your signup.

You can also look in the auth.users table in the dashboard table UI (select auth schema) and see what columns and data is available to use in the trigger function.

You could also show your function and someone might be able to point out your syntax error or help further with the issue.

**Reply by xt42io** [reply]
I  just edited the post, I have included the code, please have a look.

**Reply by GaryAustin1** [reply]
For sure your `new.raw_user_meta_data.name` is wrong and should be `->>'name'` IF THAT EVEN EXISTS.
Edit:  Also you are getting email from metadata and there is a new.email column that can normally be used.
Edit2: Also your order is wrong for columns and data you are providing.  email and user_name seem swapped.

It really helps to look at the log for errors to see what is wrong. 
You also might turn off the trigger (you can do that in the database trigger UI), do a signIn and look in the auth.users table raw_user_meta_data column to see what is in it for your user.  I have no idea what Discord sends in as far as user data.
